packages/in_app_purchase/lib/src/in_app_purchase_connection/app_store_connection.dart

```diff
@@ -126,12 +123,17 @@ class _TransactionObserver implements SKTransactionObserverWrapper {
 
   /// Triggered when any transactions are updated.
   void updatedTransactions({List<SKPaymentTransactionWrapper> transactions}) {
-    assert(_restoreCompleter != null);
-    _restoredTransactions =
-        transactions.where((SKPaymentTransactionWrapper wrapper) {
-      return wrapper.transactionState ==
-          SKPaymentTransactionStateWrapper.restored;
-    }).toList();
+    if (_restoreCompleter != null) {
```

Comment:
subjective: I think an early return/guard clause would make this more readable.

https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html